data/src/main/java/com/example/data/service/WeatherService.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.data.service
+
+import com.example.data.WeatherServiceRequest
+import com.example.data.mappers.transformApi
+import com.example.domain.entities.ResponseWeatherInfo
+import com.example.domain.service.WeatherInterfaceService
+import com.example.domain.utils.ServiceResult
+
+class WeatherService : WeatherInterfaceService {
+
+    private val generator = WeatherServiceRequest()
+
+    override fun getWeatherCityInfo(): ServiceResult<ResponseWeatherInfo> {
+        val callResponse =
+            generator.createService(WeatherApi::class.java).callWeatherInfo(CITY, APP_ID, UNITS)
+        try {
+            val response = callResponse.execute()
+            if (response.isSuccessful) {
+                response.body()?.transformApi()?.let { return ServiceResult.Success(it) }
+            }
+            return ServiceResult.Failure(Exception(response.message()))
+        } catch (e: Exception) {
+            return ServiceResult.Failure(Exception(e.message))
+        }
+    }
+
+    companion object {
+        private const val CITY: String = "Tandil"
+        private const val APP_ID: String = "b6243258fce26de557248d662c2aef4c"
```

Comment:
please add this ID to a grradle properti

Reply:
Doing

Reply:
for this, you should use a buildConfigField and call the real APP_ID from the global properties, the key should not be hardcoded in the gradle

Reply:
Done